ot3_state_manager/ot3_state_manager/messages.py

```diff
@@ -52,55 +46,94 @@ class MoveMessage(Message):
     axis: OT3Axis
     direction: Direction
 
-    @classmethod
-    def to_message(cls, string: str) -> Optional[MoveMessage]:
-        """Parse string to Message or None."""
-        split_string = string.split(" ")
-
-        if len(split_string) == 2:
-            try:
-                axis = OT3Axis[split_string[1]]
-            except KeyError:
-                axis = None
-
-            try:
-                direction = Direction.from_symbol(split_string[0])
-            except ValueError:
-                direction = None
-
-            if axis is not None and direction is not None:
-                return cls(axis=axis, direction=direction)
+    def __eq__(self, other: Any) -> bool:
+        """Confirm that the 2 objects are equal"""
+        return isinstance(other, MoveMessage) and (
+            self.axis == other.axis and self.direction == other.direction
+        )
+
+    def __hash__(self) -> int:
+        """Return hash for instance."""
+        return hash(str(self))
+
+
+class MessageLookup:
+    """Lookup class for mapping a message to a byte value."""
+
+    MESSAGE_LOOKUP_DICT: Dict[int, Message] = {
+        0x0000: MoveMessage(OT3Axis.X, Direction.NEGATIVE),
```

Comment:
rather than having the entirety of a binary message be a key in this dict, let's separate this into layers:
- Define the message binary format as having a `message-id` part (one or two bytes) that identifies the _type_ of the message, and a `message-content` part (the rest of the sequence) that contains the _content_ of the message
- Have a dict whose keys are the `message-id` part, and the values are dataclasses with a `build_from_bytes` static method that take a `bytes` instance; not dataclass instances, but the class types themselves
- Then the deserialization process is:
```
message = read_bytes(...)
message_id = struct.unpack('<H', message[:2])  # unpack the two bytes of message id
message_class = MESSAGE_LOOKUP_DICT[message_id]  # find the class
message = message_class.build_from_bytes(message[2:])
````
the dict can be typed in one of two ways:
- You can build an inheritance hierarchy for messages
- You can call it a typed dict with integer literals keys mapping to different class types. this might end up being a nightmare, fair warning